src/syntax/sedlex.ml

```diff
@@ -117,6 +129,89 @@ let transition (state : state) =
   Array.sort (fun (c1, _) (c2, _) -> compare c1 c2) t;
   t
 
+(* Restore NFA path from DFA path *)
+
+type trans_case = {
+  curr_state : int;
+  curr_node : int;
+  prev_state : int;
+  prev_node : int;
+  char_set : Sedlex_cset.t;
+  actions : action list;
+}
+
+type final_case = { curr_node : int; actions : action list }
+
+let compile_traces states (start, final) =
+  let counter = ref 0 in
+  let nodes_idx = Hashtbl.create 31 in
```

Comment:
How are `nodes_idx` different from `node.id` ?

Reply:
Right. I forgot there's already an id in node.

Reply:
Addressed